lib/src/mapbox_map.dart

```diff
@@ -426,11 +426,43 @@ class MapboxMap extends ChangeNotifier {
           _RenderedQueryGeometry(value: geometry.value, type: geometry.type),
           options);
 
+  /// Queries the map for rendered features using featureset descriptors.
+  @experimental
+  Future<List<QueriedRenderedFeature?>> queryRenderedFeaturesForTargets(
+          RenderedQueryGeometry geometry,
+          List<FeaturesetQueryTarget> targets) async =>
+      _mapInterface.queryRenderedFeaturesForTargets(
+          _RenderedQueryGeometry(value: geometry.value, type: geometry.type),
+          targets);
+
+  /// Queries the map for rendered features with one typed featureset.
+  @experimental
+  Future<List<FeaturesetFeature>> queryRenderedFeaturesForFeatureset(
```

Comment:
The big part of iOS and Android SDK is the introduction of generated types specific for StandardStyle, and the generic-based system for this. The similar call in IOS leverages this system and returns you a feature type with corresponding type, e.g something like this:

```
let poiFeatures = await map.queryRenderedFeaturesForFeatureset(geom, FeaturesetDescriptor.standardPoi)
// poiFeatures here is a List<StandardPoiFeature>
```

If we can express this in Dart, I'd recommend doing this because it makes the api much more usable. WDYT?

Reply:
> If we can express this in Dart

That's what I'm trying to figure out in the ongoing work: how to express this in Dart in a way that is (1) handles all necessary conversions across the platforms, (2) can be generated by our pigeons, and (3) creates a consistent public API with the iOS and Android APIs.